openedx/core/lib/request_utils.py

```diff
@@ -227,3 +228,29 @@ def set_custom_attributes_for_top_n(self, names_to_size, top_n_captured, attribu
             set_custom_attribute(name_attribute, name)
             set_custom_attribute(size_attribute, size)
             log.debug('%s = %d', name, size)
+
+# .. toggle_name: ENABLE_403_MONITORING
+# .. toggle_implementation: SettingToggle
+# .. toggle_default: False
+# .. toggle_description: Temporary toggle to track down the source of 403s for /oauth2/exchange_access_token/.
+# .. toggle_use_cases: temporary
+# .. toggle_creation_date: 2021-02-12
+# .. toggle_target_removal_date: 2021-03-12
+# .. toggle_tickets: https://openedx.atlassian.net/browse/ARCHBOM-1667
+ENABLE_403_MONITORING = SettingToggle('ENABLE_403_MONITORING', default=False, module_name=__name__)
+
+
+def custom_exception_handler(exc, context):
+    """ Enables monitoring of 403s for /oauth2/exchange_access_token/ to gather data. """
+    # Call REST framework's default exception handler first,
+    # to get the standard error response.
+    response = exception_handler(exc, context)
+
+    log_403s = ENABLE_403_MONITORING.is_enabled() and response.status_code == 403
```

Comment:
@robrap This is causing some errors in prod (that are masking other errors), because [if the underlying exception is not a DRF exception, the upstream `exception_handler()` will return `None`](https://github.com/encode/django-rest-framework/blob/1ec0f86b585cd87e4b413aeaad1ecc947bacfef2/rest_framework/views.py#L71-L101) and then trying to access `response.status_code` gives `AttributeError: 'NoneType' object has no attribute 'status_code'`

Reply:
Thanks @bradenmacdonald. My first attempt was more defensive, and I forgot in this one.  However, I did add the toggle so it could have been immediately turned off (5 minute delay), but it looks like we waited for the full revert.  Oh well.  Next time.  Thanks again.